internal/server/iigointernal/judiciary.go

```diff
@@ -118,7 +140,96 @@ func (j *judiciary) appointNextPresident(currentPresident shared.ClientID, allIs
 	return nextPresident
 }
 
-// cycleSanctionCache rolls the sanction cache one turn forward (effectively dropping any sanctions longer than the depth)
+// updateSanctionScore uses results of InspectHistory to assign sanction scores to clients
+// This function relies upon (in part) the client provided RuleViolationSeverity
+func (j *judiciary) updateSanctionScore() {
+	//Clearing sanction map
+	j.sanctionRecord = map[shared.ClientID]roles.IIGOSanctionScore{}
+	islandTransgressions := map[shared.ClientID][]string{}
+	for k, v := range j.evaluationResults {
+		islandTransgressions[k] = unpackSingleIslandTransgressions(v)
+	}
+	j.scoreIslandTransgressions(islandTransgressions)
+}
+
+// scoreIslandTransgressions uses the client provided ruleViolationSeverity map to score island transgressions
+func (j *judiciary) scoreIslandTransgressions(transgressions map[shared.ClientID][]string) {
+	for islandId, rulesBroken := range transgressions {
+		totalIslandTurnScore := roles.IIGOSanctionScore(0)
+		for _, ruleBroken := range rulesBroken {
+			if score, ok := j.ruleViolationSeverity[ruleBroken]; ok {
+				totalIslandTurnScore += score
+			} else {
+				totalIslandTurnScore += roles.IIGOSanctionScore(1)
+			}
+		}
+		j.sanctionRecord[islandId] += totalIslandTurnScore
+	}
+}
+
+// applySanctions uses RulesInPlay and it's versions of the sanction rules to work out how much to sanction an island
+func (j *judiciary) applySanctions() {
+	j.cycleSanctionCache()
+	var currentSanctions []roles.Sanction
+	for islandID, sanctionScore := range j.sanctionRecord {
+		islandSanctionTier := getIslandSanctionTier(sanctionScore, j.sanctionThresholds)
+		sanctionEntry := roles.Sanction{
+			ClientID:     islandID,
+			SanctionTier: islandSanctionTier,
+			TurnsLeft:    sanctionLength,
+		}
+		currentSanctions = append(currentSanctions, sanctionEntry)
+	}
+	j.localSanctionCache[0] = currentSanctions
```

Comment:
what if out of range

Reply:
The first line of the applySanctions function: "j.cycleSanctionCache()" (line 170), rebuilds the sanction cache and puts in an empty entry into j.localSanctionCache[0], this should mean that it isn't out of range.

Reply:
ok, i trust you to remember this one day before the deadline...

Reply:
Haha, on my head be it. Tbf the localSanctionCache is an instantiated map anyway so at worst this is just assigning a value to a new key (even if it shouldn't be new)?

Reply:
ah ok - it's a map. what does the key represent and why is it 0 here?

Reply:
The key represents how many timesteps ago the sanction was levied. The 0 was since the sanctions assigned here are ones that came from rules that were evaluated in the current session of IIGO.

Reply:
I believe it represents turns since the sanctions have come into play, cycleSanctionCache moves this one up until the max turns a sanction can be a in play at which point it is removed